### PR TITLE
fix(gux-form-field-color): added border styles

### PR DESCRIPTION
**Pull request by jason-evans-genesys**
https://inindca.atlassian.net/browse/COMUI-2406

Added border styling.

### CODE REVIEW

File: packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.scss
Comment (MattCheely) [inline packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.scss:66]: This isn't a suitable fix, because it breaks the aspect ratio of the component - it's no longer a square. You can see this in the layout tab of the CSS inspect, which is always a good place to look when you run into this kind of issue:

Screen Shot 2023-11-03 at 9 04 28 AM

The design also calls for a 32x32 square, rather than 34x34. When you're having trouble with an element's size, it's often a good idea to check it's children. In this case, the slotted input is also out of square, which is likely part of the problem:

Screen Shot 2023-11-03 at 9 08 48 AM

The input is bigger than I expected as well at 32x32px instead of the 24x24 swatch from the designs. After removing the extra 2px on this line, I was able to get the correct sizing with these styles on the input:

```css
::slotted(input) {
  ...
  flex: 0 1 auto; //don't grow to the container, we want a fixed width
  width: 24px; // swatch size from designs
  height: 24px;
  padding: 0; // get rid of the browser padding
  ...
}
```

**Review by MattCheely — CHANGES_REQUESTED** [review]

File: packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.scss
Comment (MattCheely) [reply packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.scss:66]: Edit: Looks like those styles work in Firefox, but maybe not in chrome. It may take some more digging to find a solution for both, or Chrome may need some additional overrides for browser CSS.

Comment (MattCheely) [reply packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.scss:66]: It might be worth looking at [-moz-color-swatch](https://developer.mozilla.org/en-US/docs/Web/CSS/::-moz-color-swatch) and -webkit-color-swatch as options to get the styles we want.

Comment (jason-evans-genesys) [reply packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.scss]: Thanks, Matt for this really helpful feedback and great catches. The swatch and its container looks to render better in both chrome and firefox, but I'm still working on a few more issues with the component, such as the focus ring styling, and getting the aspect ratio to be 24x24 for the swatch, instead of 22x24 right now. I'll let you know when this is ready for review again.

File: packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.scss
Comment (jason-evans-genesys) [inline packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.scss:24]: I added this min-width so that the aspect ratio can be 24x24 for the swatch. Without the min-width the swatch aspect ratio is 22.2x24 (I believe this is because the browser automatically sets a default width to the input, so this min-width style is a way to override that)

File: packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.light.scss
Comment (jason-evans-genesys) [inline packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.light.scss:4]: This focus styling here will prevent 2 focus rings from rendering - we only want one focus ring on the outer container to render.

File: packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.scss
Comment (jason-evans-genesys) [reply packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.scss]: Thanks for the very helpful feedback @MattCheely 

I pushed up changes for this.

File: packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.light.scss
Comment (MattCheely) [inline packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.light.scss:17]: This is good, but now that we can style the swatch, you can get a little closer to the designs by also adding:

```
  border: none;
  border-radius: var(--gse-ui-colorPicker-input-swatchBorderRadius);
```

**Review by MattCheely — CHANGES_REQUESTED** [review]

File: packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.light.scss
Comment (MattCheely) [reply packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.light.scss:17]: Once this change is in, let's ask the UX team to review it before we merge.

Comment (jason-evans-genesys) [reply packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.light.scss:17]: @MattCheely I got the approval from the UX team.

Comment (jason-evans-genesys) [reply packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.light.scss:17]: There's currently a conflict on style.scss, but I was thinking I could resolve this conflict when I do a git rebase when I merge this PR after it's approved.

**Comment by daragh-king-genesys** [comment]
you will have to rebase this on main. It should be something like this.
```
git checkout main
git fetch --prune
git pull
git checkout feature/COMUI-2406
git rebase main
git push --force
```

File: packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.light.scss
Comment (jason-evans-genesys) [reply packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-color/gux-form-field-color.light.scss:17]: @MattCheely Just wanted to check in and see if the changes I pushed for this PR look ok and this PR is ready for approval. Thanks!

**Review by MattCheely — APPROVED** [review]
👍 Looks good!

**Pull request merged**
